src/spikeinterface/core/core_tools.py

```diff
@@ -163,7 +163,7 @@ def make_shared_array(shape, dtype):
     from multiprocessing.shared_memory import SharedMemory
 
     dtype = np.dtype(dtype)
-    nbytes = int(np.prod(shape) * dtype.itemsize)
+    nbytes = prod(shape) * dtype.itemsize
```

Comment:
This still needs to be wrapped as an int no?

Reply:
Related to Windows failing.

Reply:
itemsize is int, shape is int, that's unecessary. Am I wrong?

Reply:
When I check they are coming up as ints, but based on the fact the tests are changing just for Windows and all you did was remove the int over the operation maybe something was being converted.... Not sure.

Reply:
The thing that converts is the `np.prod`. because of an overflow. Let me show you:

```python
In [1]: a_billion_int = 1_000_000_000

In [2]: np.prod((a_billion_int, a_billion_int, a_billion_int, a_billion_int))
Out[2]: -5527149226598858752
```

Reply:
Agreed. np.prod was causing an overflow. Absolutely agree, but now tests are failing.

So why do you thing nbytes are coming up as np.int for Windows testing only?

Reply:
I have no idea, my friend. There must be a casting somewhere else.

Reply:
Yes, so TO MY SURPRISE `np.prod` does preserve casting:

```
In [3]: from math import prod

In [4]: x = np.arange(9, dtype="int")[8]

In [5]: prod((x, x))
Out[5]: 64

In [6]: type(prod((x, x)))
Out[6]: numpy.int64
```

(The beauty and dangers of duck-type)!

To realy solve the bug we need to figure out which shape comes as numpy scalars. Otherwise, even this solution will break at some ponint.

Reply:
That's wild.

Reply:
Which points to the solution. There is a place where a numpy functions were used to determine the number of channels so we ended up with a numpy scalar on shape. I changed that but I guess the most robus solution is to force casting **inside** the shape. That avoids the error from the root whereas casting the final product (already overflown and hence negative) would not.

Does this makes sense?

Reply:
I considered the problem solved but I still don't know why the first iteraiton of the solution did fail on windows and not on unbutu : O

Reply:
It's a good thing it did fail because it found another potential point of overflow. I know numpy optimization is different between different OSs so maybe that did it on Windows? But I don't think the different OSs should return different types.... Another one of the great mysteries.

Reply:
Windows and linux numpy handle overflow differently. [See this](https://github.com/numpy/numpy/issues/8433). Based on the discussion the behavior will likely be changed in Numpy 2.0, but in short
```
np.full((2, 2), np.iinfo(np.int32).max, dtype=np.int32).sum(axis=0)
shows similar behavior: array([4294967294, 4294967294]) on ubuntu and array([-2, -2]) on windows.
```
> However, this function returns an array instead of a scalar, so we can inspect the dtype. On Ubuntu the dtype is int64, but on Windows the dtype is int32.

So maybe this was due to the difference in how certain cases of overflow occur.

Reply:
Thanks a bunch for sharing the issue!

Reply:
So numpy 2.0 should align this.

Reply:
Anything that makes the debugging easier among the operating systems. Much less surface area if one fixes works for all of them. :) (although the final message wasn't completely confident sounding...)